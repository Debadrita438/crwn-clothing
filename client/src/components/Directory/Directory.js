import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectDirectorySections } from '../../redux/directory/directorySelector';

import MenuItem from '../MenuItem/MenuItem';

import { DirectoryMenuContainer } from './Directory.styles';


const mapStateToProps = createStructuredSelector({
    sections: selectDirectorySections
});

export const Directory = ({ sections }) => {
  return (
      <DirectoryMenuContainer>
          {
              sections.map(({  id, ...otherSectionProps }) => (
                  <MenuItem key={ id } { ...otherSectionProps } />
              ))
          }
      </DirectoryMenuContainer>
  );
}

 
export default connect(mapStateToProps)(Directory);