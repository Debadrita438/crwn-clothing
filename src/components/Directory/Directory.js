import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectDirectorySections } from '../../redux/directory/directorySelector';

import MenuItem from '../MenuItem/MenuItem';

import './Directory.scss';


const mapStateToProps = createStructuredSelector({
    sections: selectDirectorySections
});

const Directory = ({ sections }) => {
  return (
      <div className="directory-menu">
          {
              sections.map(({  id, ...otherSectionProps }) => (
                  <MenuItem key={ id } { ...otherSectionProps } />
              ))
          }
      </div>
  );
}

 
export default connect(mapStateToProps)(Directory);