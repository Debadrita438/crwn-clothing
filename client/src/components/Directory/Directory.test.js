import { shallow } from 'enzyme';

import { Directory } from './Directory';

it('should render Directory Component', () => {
    expect(shallow(<Directory sections={[]} />)).toMatchSnapshot();
});