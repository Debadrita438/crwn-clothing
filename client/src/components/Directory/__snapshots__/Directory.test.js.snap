// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Directory Component 1`] = `ShallowWrapper {}`;
